### PR TITLE
Feature LTI 1.1 und 1.3

### PR DESCRIPTION
**Pull request by dominikglandorf**
Adds a plugin, usable by setting environment variable ENABLE_LTI=1, that reads available LTI tools from the schulcloud-server and saves the choice and displays the tool in the editor.

Without changes on schulcloud-server LTI1.3 is not available. This requires https://github.com/schul-cloud/schulcloud-server/pull/942

### CODE REVIEW

File: src/workspace/Lesson/Section/plugins.js [hunk 1]
```diff
@@ -99,4 +100,6 @@ const plugins = {
 	// h5p: h5pPlugin,
 }
 
+if(process.env.ENABLE_LTI) plugins['lti'] = ltiPlugin;
```
Comment (CeEv) [inline src/workspace/Lesson/Section/plugins.js]: Please use the src/config.js file to set ENABLE_LTI.

File: src/plugins/lti/LTI.jsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          },
+          'https://purl.imsglobal.org/spec/lti/claim/version': state.ltiVersion.value,
+          'https://purl.imsglobal.org/spec/lti/claim/deployment_id': state.id.value,
+          'https://purl.imsglobal.org/spec/lti-dl/claim/deep_linking_settings':
+            (state.ltiMessageType.value === 'LtiDeepLinkingRequest'
+              ? {
+                accept_types: ['ltiLink'],
+                accept_media_types: 'image/*,text/html',
+                accept_presentation_document_targets: ['iframe', 'window'],
+                deep_link_return_url: `http://localhost:3030/tools/${state.id.value}/link`,
```
Comment (CeEv) [inline src/plugins/lti/LTI.jsx]: http://localhost:3030 take the value from the src/config.js  -> SERVER_API_URL

File: src/plugins/lti/LTI.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          iss,
+          name,
+          aud: state.clientId.value,
+          sub: pseudonym,
+          exp: current.getTime() + 3 * 60,
+          iat: current.getTime(),
+          nonce: shortid.generate() + shortid.generate(),
+          'https://purl.imsglobal.org/spec/lti/claim/message_type': state.ltiMessageType.value,
+          'https://purl.imsglobal.org/spec/lti/claim/roles': [
+            `http://purl.imsglobal.org/vocab/lis/v2/membership#${user.roles[0].name}`,
```
Comment (CeEv) [inline src/plugins/lti/LTI.jsx]: It exist situations where the user can have more then one role. Is it important?

File: src/plugins/lti/LTI.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+  const setTool = (tool) => {
+    state.id.set(shortid.generate())
+    state.templateId.set(tool._id)
+    state.clientId.set(tool.oAuthClientId)
+    state.ltiMessageType.set(tool.lti_message_type)
+    state.ltiVersion.set(tool.lti_version)
+    state.url.set(tool.url)
+  }
+
+  if (loading) {
```
Comment (CeEv) [inline src/plugins/lti/LTI.jsx]: Only style, but maybe the if / else render is more readable if you use named helper for the render block.
const renderDefault = () => return ( <div> ..
if() renderDefault();  ...

**Comment by janrenz** [comment]
@dominikglandorf can you check Cedrics comments

File: src/plugins/lti/LTI.jsx (hunk 4 above)
Comment (dominikglandorf) [reply src/plugins/lti/LTI.jsx]: Thanks!

**Comment by janrenz** [comment]
@dominikglandorf ping ...

**Comment by dominikglandorf** [comment]
@janrenz It works now

**Comment by janrenz** [comment]
@dominikglandorf Please fix conflicting files

**Comment by dominikglandorf** [comment]
@janrenz Not working right now, needs adaption to new plugin structure. Coming soon.

**Comment by CeEv** [comment]
@dominikglandorf  please look at https://github.com/schul-cloud/edtrio/blob/develop/src/plugins/index.js. It should help.
Please write me in rc if you ready or need more informations. I want to merge it as quickly as possible, before more breaking chances, or conflicts are coming.

**Pull request merged**